shared/tests/claimify/test_integration.py

```diff
@@ -0,0 +1,534 @@
+        """Set up test data and integration instance."""
+        if not NEO4J_AVAILABLE:
+            self.skipTest("Neo4j dependencies not available")
+            
+        # Mock the neo4j dependency if not available
+        try:
+            from ..graph.neo4j_manager import Neo4jGraphManager
+            # In tests, we'll use a mocked manager
+            self.graph_manager = MockNeo4jGraphManager()
+        except ImportError:
+            # If neo4j is not available, use the mock
+            self.graph_manager = MockNeo4jGraphManager()
+        
+        self.integration = ClaimifyGraphIntegration(self.graph_manager)
+        
+        # Create test sentence chunks
+        self.test_chunk = SentenceChunk(
+            text="The system failed at startup.",
+            source_id="blk_001",
+            chunk_id="chunk_001",
+            sentence_index=0
+        )
+        
+        self.test_context = ClaimifyContext(current_sentence=self.test_chunk)
+    
+    def test_integration_initialization(self):
+        """Test integration initialization."""
+        integration = ClaimifyGraphIntegration(self.graph_manager)
+        self.assertEqual(integration.graph_manager, self.graph_manager)
+    
+    def test_convert_successful_claim_result(self):
+        """Test conversion of result with valid claims."""
+        # Create a successful processing result with valid claims
+        claim_candidate = ClaimCandidate(
+            text="The system failed at startup.",
+            is_atomic=True,
+            is_self_contained=True,
+            is_verifiable=True,
+            confidence=0.9
+        )
+        
+        decomposition_result = DecompositionResult(
+            original_text="The system failed at startup.",
+            claim_candidates=[claim_candidate]
+        )
+        
+        result = ClaimifyResult(
+            original_chunk=self.test_chunk,
+            context=self.test_context,
+            selection_result=SelectionResult(self.test_chunk, is_selected=True),
+            disambiguation_result=DisambiguationResult(
+                self.test_chunk, 
+                "The system failed at startup."
+            ),
+            decomposition_result=decomposition_result
+        )
+        
+        # Convert to graph inputs
+        claim_inputs, sentence_inputs = self.integration._convert_result_to_inputs(result)
+        
+        # Should have one claim, no sentences
+        self.assertEqual(len(claim_inputs), 1)
+        self.assertEqual(len(sentence_inputs), 0)
+        
+        # Check claim properties
+        claim = claim_inputs[0]
+        self.assertIsInstance(claim, ClaimInput)
+        self.assertEqual(claim.text, "The system failed at startup.")
+        self.assertEqual(claim.block_id, "blk_001")
+        self.assertTrue(claim.verifiable)
+        self.assertTrue(claim.self_contained)
+        self.assertTrue(claim.context_complete)
+    
+    def test_convert_failed_claim_result(self):
```

Comment:
________ TestClaimifyGraphIntegration.test_convert_failed_claim_result _________
shared/tests/claimify/test_integration.py:188: in test_convert_failed_claim_result
    self.assertFalse(sentence.failed_decomposition)  # Was atomic
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   AttributeError: 'SentenceInput' object has no attribute 'failed_decomposition'